src/backend/logging/backend_logger.h

```diff
@@ -36,34 +41,93 @@ class BackendLogger : public Logger {
 
   static BackendLogger *GetBackendLogger(LoggingType logging_type);
 
-  void FinishedFlushing(void);
-
-  void WaitForFlushing(void);
-
   //===--------------------------------------------------------------------===//
   // Virtual Functions
   //===--------------------------------------------------------------------===//
 
   // Log the given record
-  virtual void Log(LogRecord *record) = 0;
+  void Log(LogRecord *record);
 
   // Construct a log record with tuple information
   virtual LogRecord *GetTupleRecord(LogRecordType log_record_type,
-                                    txn_id_t txn_id,
-                                    oid_t table_oid,
-                                    oid_t db_oid,
-                                    ItemPointer insert_location,
+                                    txn_id_t txn_id, oid_t table_oid,
+                                    oid_t db_oid, ItemPointer insert_location,
                                     ItemPointer delete_location,
                                     const void *data = nullptr) = 0;
 
+  void SetLoggingCidLowerBound(cid_t cid) {
+    // XXX bad synchronization practice
+    log_buffer_lock.Lock();
+
+    logging_cid_lower_bound = cid;
+
+    highest_logged_commit_message = INVALID_CID;
+
+    log_buffer_lock.Unlock();
+  }
+
+  // FIXME The following methods should be exposed to FrontendLogger only
+  // Collect all log buffers to be persisted
+  std::vector<std::unique_ptr<LogBuffer>> &GetLogBuffers() {
```

Comment:
This method returns a non-const reference of a member variable. Is this intended?